lib/order.rb

```diff
@@ -0,0 +1,85 @@
+require_relative "sms"
+
+class Order
+  attr_reader :id, :restaurant, :total, :items
+
+  @@orders = []
+
+  def initialize(restaurant)
+    @id = 1 + @@orders.count
+    @@orders.push(self)
+    @restaurant = restaurant
+    @total = 0
+    @items = []
+  end
+
+  def add_item(item, quantity = 1)
+    raise "This item doesn't exist at this restaurant." if get_item(item).nil?
+    increase_items(item, quantity)
+    increase_total(quantity)
```

Comment:
I think there are two problems with this:

1) It's not clear what it does. The name mentions total but you pass it a quantity - it suggests that total is the total number of items, not the order total.

2) When I examine the code, I see that it is tightly coupled with increase_items i.e. it only works as expected if increase_items has been called immediately before.

Reply:
1 - Is this a case of renaming the variable you think? Or is there something deeper that makes this hard to understand?

2 - Is that a problem? Considering they are both private methods and I am calling them successively? In any case this would be removed by your awesome point of removing the @ total variable!